tests/test_deployments.py

```diff
@@ -946,6 +964,74 @@ async def test_deployment_apply_syncs_triggers(
                     create_route.calls[0].request.content
                 ) == trigger.as_automation().dict(json_compatible=True)
 
+    @pytest.mark.parametrize("enable_flag", [True, False])
+    async def test_trigger_job_vars_value_if_enable_infra_overrides_flag_is_toggled(
+        self, patch_import, tmp_path, enable_flag: bool
+    ):
+        infrastructure = Process()
+        await infrastructure._save(is_anonymous=True)
+
+        trigger = pydantic.parse_obj_as(
+            DeploymentTriggerTypes, {"job_variables": {"foo": 123}}
+        )
+
+        deployment = Deployment(
+            name="TEST",
+            flow_name="fn",
+            triggers=[trigger],
+            infrastructure=infrastructure,
+        )
+
+        created_deployment_id = str(uuid4())
+
+        updates = {
+            PREFECT_API_URL: f"https://api.prefect.cloud/api/accounts/{uuid4()}/workspaces/{uuid4()}",
+            PREFECT_CLOUD_API_URL: "https://api.prefect.cloud/api/",
+        }
+        if enable_flag:
+            updates[PREFECT_EXPERIMENTAL_ENABLE_FLOW_RUN_INFRA_OVERRIDES] = True
+
+        with temporary_settings(updates=updates):
```

Comment:
Let me know how this works out, I've had some trouble using `with temporary_settings` inside of `async` functions.  If you hit any snags with this, you'll need to pull them up to a sync fixture and `yield` out of it

Reply:
Interesting. I haven't hit that yet, but I'll keep an eye out. Is the failure recognizable or does it show up as a timeout, etc.?

Reply:
I think it shows up as logging noise about `ContextVar`s?